dsgrid/config/dimensions.py

```diff
@@ -139,6 +149,22 @@ def check_name(cls, name):
             )
         return name
 
+    @validator("display_name")
+    def check_display_name(cls, display_name):
+        if display_name == "":
+            raise ValueError(f'Empty name field for dimension: "{cls}"')
+        if REGEX_VALID_REGISTRY_DISPLAY_NAME.search(display_name) is None:
+            raise ValueError(f"display_name={display_name} does not meet the requirements")
+        return display_name
+
+    @validator("query_name")
+    def check_query_name(cls, query_name, values):
+        if "display_name" not in values:
+            return query_name
+        if query_name is not None:
+            return query_name
+        return values["display_name"].lower().replace(" ", "_")
```

Comment:
I see what you did here, :)

Reply:
We do. In this regular expression `\w` allows letters, numbers, and underscores. There is probably some debate about whether underscore is alphanumeric. In Python it is.
```
REGEX_VALID_REGISTRY_DISPLAY_NAME = re.compile(r"^[\w ]+$")
```

Reply:
I am wondering if we should support dashes and convert them to underscores in the query name. Personally, I'd rather not have dashes or underscores in display names, but if we do have them I would prefer dashes to underscores